src/mesh/parallel/fci.cxx

```diff
@@ -251,17 +312,21 @@ FCIMap::FCIMap(Mesh& mesh, Options& options, int offset_, BoundaryRegionPar* bou
           // Invert 2x2 matrix to get change in index
           BoutReal dx = (dZ_dz * dR - dR_dz * dZ) / det;
           BoutReal dz = (dR_dx * dZ - dZ_dx * dR) / det;
-          boundary->add_point(x, y, z,
-                              x + dx, y + 0.5*offset, z + dz,  // Intersection point in local index space
-                              0.5*coord.dy(x,y), //sqrt( SQ(dR) + SQ(dZ) ),  // Distance to intersection
-                              PI   // Right-angle intersection
-                              );
+          outer_boundary->add_point(
+              x, y, z, x + dx, y + 0.5 * offset,
+              z + dz, // Intersection point in local index space
+              0.5
+                  * coord.dy(x, y,
+                             z), // sqrt( SQ(dR) + SQ(dZ) ),  // Distance to intersection
+              PI                 // Right-angle intersection
+          );
         }
       }
     }
   }
 
   interp->setMask(boundary_mask);
+  interp->calcWeights(abs(xt_prime), zt_prime);
```

Comment:
I had some question about using `abs` in a previous review. I'm suspicious of the `abs(xt_prime)` here too, but don't understand what's being done to know if it's a real issue.